include/private/entry.h

```diff
@@ -122,40 +122,101 @@ locked_add_element( struct stumpless_entry *entry,
  *
  * **Async Signal Safety: AS-Unsafe**
- * This function is not safe to call from signal handlers due to use of
- * thread global structrues.
+ * Not safe to call from asynchronous signal handlers due to potential shared data access.
  *
  * **Async Cancel Safety: AC-Unsafe**
- * This function is not safe to call from threads that may be asynchronously 
- * canceled due to use of thread-global structure that may left in a
- * undefined state when cancelled.
+ * Not safe in contexts of asynchronous cancellation, as it might involve unsafe data access.
  *
- * @param entry The entry to get the element from.
+ * @since release 2.0.0
  *
- * @param index The index of element within the entry.
+ * @param entry The entry containing the elements. Must not be NULL.
+ * @param index The index of the element to retrieve.
  *
- * @return The element if no error is encountered. If an error is
- * encountered, then NULL is returned and an error code is set appropriately.
+ * @return A pointer to the element at the specified index, or NULL if not found or in case of an error.
  */
 struct stumpless_element *
 locked_get_element_by_index( const struct stumpless_entry *entry,
                              size_t index );
 
+/**
+ * Searches for an element by name within a given Stumpless entry, assuming external thread safety management.
+ *
+ * This function looks for an element with the specified name in the provided entry.
+ * It does not manage thread safety internally but assumes that the calling function
+ * has already acquired the necessary locks on the entry.
+ *
+ * **Thread Safety: MT-Unsafe**
+ * The function itself is not thread-safe. It relies on the caller to ensure thread safety
+ * by appropriately locking the entry before calling.
+ *
+ * **Async Signal Safety: AS-Unsafe**
+ * Unsafe to call from asynchronous signal handlers due to potential shared data access.
+ *
+ * **Async Cancel Safety: AC-Unsafe**
+ * Not safe in contexts of asynchronous cancellation, as it might involve unsafe data access.
+ *
+ * @since release 2.0.0
+ *
+ * @param entry The entry containing the elements to be searched. Must not be NULL.
+ * @param name The name of the element to find. Must be a NULL-terminated string.
+ *
+ * @return A pointer to the found element, or NULL if not found or in case of an error.
+ */
 struct stumpless_element *
 locked_get_element_by_name( const struct stumpless_entry *entry,
                             const char *name );
 
 /**
- * Creates a new entry with the given parameters.
+ * Creates a new Stumpless entry with specified parameters.
+ *
+ * Allocates a new entry and initializes it with the provided facility,
+ * severity, application name, message ID, and message. It assumes that memory
+ * allocation (e.g., for the entry itself and string copies) is thread-safe as
+ * provided by the system's standard library.
+ */**
+ * Locks the mutex within the entry.
```

Comment:
It looks like the documentation for `lock_entry` and `create_entry` got interleaved somehow?